src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -450,6 +464,12 @@ double Helix::safePitch()
     // Below is an approximation. It is possible to do the general way by solving for the pitch
     // where the helix is self intersecting.
 
+    Base::Vector3d v = Axis.getValue();
+    Base::Vector3d n = getProfileNormal();
+    Base::Vector3d s = v.Cross(n);  // pointing towards the desired helix start point.
+    if (s.IsNull())
+        return 0.0; // if the axis orthogonal to the profile, any pitch >0 is safe
```

Comment:
It's inconsistent between the comment and the returned value.

Reply:
Indeed, at the (only) usage of this "safePitch" value, in `TaskHelixParameters::updateUI()`, it is compared against the current pitch value like this:
```cpp
        if (pcHelix->safePitch() > propPitch->getValue())
            status = "Warning: helix might be self intersecting";
```
So an actual pitch value of exactly 0.0 can slip though, i.e. without a warning status message. A zero pitch is used then creating a flat spiral, but this again doesn't make any sense with an axis normal to the profile.

Would `Precision::Confusion()` be a sensible value to return instead?

Reply:
Actually I don't know which value is sensible. Maybe 0.0 is. Just do not at the same time 'return 0.0' and comment 'any value strictly greater than 0 is OK'. ;)

Reply:
I decided to go for `Precision::Confusion()`. The alternative, i.e. returning 0.0 and change the usage from `>` to `>=` (and maybe rename to `safePitchLimit`), seems more risky to me.